lib/__tests__/has-style.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-env jest */
+
+import TestAssertions from "../helpers/test-assertions";
+
+describe('assert.dom(...).hasStyle()', () => {
+  let assert;
+
+  beforeEach(() => {
+    assert = new TestAssertions();
+    document.body.innerHTML = '<div class="foo" style="opacity: 1; width: 200px; text-align: center;">quit-dom ftw!</div>';
+  });
+
+  test('succeeds for correct content', () => {
+    assert.dom('.foo').hasStyle({
+      opacity: '1',
+      width: '200px',
+      'text-align': 'center',
+    });
+    expect(assert.results).toEqual([{
+      actual: { opacity: '1', width: '200px', 'text-align': 'center' },
+      expected: { opacity: '1', width: '200px', 'text-align': 'center' },
+      message: 'Element .foo has style \"{\"opacity\":\"1\",\"width\":\"200px\",\"text-align\":\"center\"}\"',
+      result: true,
+    }]);
+  });
```

Comment:
can you add another test for a partial check? right now the element has three properties, and we check exactly those properties, but we should make sure that it also works if we only check a subset of those properties.

I'm not sure if jsdom can do that but we should also test what happens if we define the `.foo` CSS class, assign properties in there and then check them with `hasStyle()`